File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
        string word="";
        vector<string> words;

        for(char c:s)
        {
            if(c!=' ') word+=c;
            else if(!word.empty()) 
            {
                 words.push_back(word);
                 word="";
            }
        }
        if(!word.empty()) words.push_back(word);

        string result="";
        for(int i=words.size()-1;i>=0;i--)
        {
            result+=words[i];
            if(i!=0) result+=' ';
        }
        return result;
    }
};

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
    String[] words = s.split(" ");
    String result = "";

    for(int i=words.length-1;i>=0;i--)
    {
        if (words[i].length() == 0) continue;
        if (result.length() == 0)
                result = words[i];
        else result=result+" "+words[i];
        //if(i!=0) result+=" ";
    }
    return result;
    }
}